storey/table.py

```diff
@@ -97,8 +141,10 @@ def _new_aggregated_store_element(self):
 
     async def add_aggregation_by_key(self, key, base_timestamp, initial_data):
         if not self._schema:
-            await self._get_or_save_schema()
-        self._set_aggregations_attrs(key, self._new_aggregated_store_element()(key, self._aggregates, base_timestamp, initial_data))
+            async with self._get_schema_lock():
```

Comment:
this function is called only from _lazy_load_key_with_aggregates and it will be already locked from there